tools/swift-ide-test/swift-ide-test.cpp

```diff
@@ -677,6 +681,66 @@ removeCodeCompletionTokens(llvm::MemoryBuffer *Input,
                                            Input->getBufferIdentifier()));
 }
 
+static int doContextInfo(const CompilerInvocation &InitInvok,
+                         StringRef SourceFilename,
+                         StringRef SecondSourceFileName,
+                         StringRef CodeCompletionToken,
+                         bool CodeCompletionDiagnostics) {
+  llvm::ErrorOr<std::unique_ptr<llvm::MemoryBuffer>> FileBufOrErr =
+    llvm::MemoryBuffer::getFile(SourceFilename);
+  if (!FileBufOrErr) {
+    llvm::errs() << "error opening input file: "
+                 << FileBufOrErr.getError().message() << '\n';
+    return 1;
+  }
+
+  unsigned Offset;
+
+  std::unique_ptr<llvm::MemoryBuffer> CleanFile(
+      removeCodeCompletionTokens(FileBufOrErr.get().get(), CodeCompletionToken,
+                                 &Offset));
+
+  if (Offset == ~0U) {
+    llvm::errs() << "could not find code completion token \""
+                 << CodeCompletionToken << "\"\n";
+    return 1;
+  }
+  llvm::outs() << "found code completion token " << CodeCompletionToken
+               << " at offset " << Offset << "\n";
+  llvm::errs() << "found code completion token " << CodeCompletionToken
```

Comment:
Huh, I wonder why we write this to both `outs()` and `errs()`.  Seems silly.  I guess it's fine to copy what code-completion does for now though.

Reply:
Yeah, I just copied from code-completion code.
I believe `outs()` is for `FileCheck` and `errs()` is to see which test case failed from failure log.